internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -398,6 +399,30 @@ func flattenTags(tags *[]admin.ResourceTag) []map[string]string {
 	return ret
 }
 
+func CheckRegionConfigsPriorityOrder(regionConfigs []admin.ReplicationSpec20240805) error {
```

Comment:
once validation is implemented in the API, can we remove this logic and only preserve the test? We could create a follow up ticket and leave a comment in the code.

Reply:
good idea, done in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2640/commits/bfb2e969d42caa25b8cabccfc2e21d0dddf71ec5